mission_control/src/mission.cpp

```diff
@@ -36,359 +36,67 @@
 // Original version: Christopher Scianna Christopher.Scianna@us.QinetiQ.com
 
 #include "mission_control/mission.h"
-#include "mission_control/behavior.h"
+
 using mission_control::Mission;
-using mission_control::Behavior;
+using namespace mission_control;
 
-Mission::Mission() : missionState(MissionState::READY), m_current_behavior(NULL), elasped_time(0.0)
+Mission::Mission()
 {
-  current_abort_behavior_id = 0;
-  current_behavior_id = 0;
+  _missionFullPath = " Hola";
+  currentBehaviorID = 0;
+  
+  _missionFactory.registerNodeType<Waypoint1>("Waypoint1");
+  _missionFactory.registerNodeType<Attitude_Servo>("Attitude_Servo");
+  _missionFactory.registerNodeType<Depth_Heading>("Depth_Heading");
```

Comment:
@patapufate meta: BehaviorTree.CPP has a mechanism to register these actions as plugins. We should aim for that.

Reply:
I'll take a look at it.

Reply:
@hidmic I'll need help with this